pkg/globalnet/controllers/cluster_egressip_controller.go

```diff
@@ -93,32 +90,52 @@ func NewClusterGlobalEgressIPController(config syncer.ResourceSyncerConfig, pool
 		return nil, err
 	}
 
+	iptIface, err := iptablesdriver.NewIPTableInterface()
+	if err != nil {
+		return nil, errors.WithMessage(err, "error creating the IPTablesInterface handler")
+	}
+
+	controller.iptIface = iptIface
+	controller.localSubnets = localSubnets
+
 	return controller, nil
 }
 
 func (c *clusterGlobalEgressIPController) process(from runtime.Object, numRequeues int, op syncer.Operation) (runtime.Object, bool) {
 	clusterGlobalEgressIP := from.(*submarinerv1.ClusterGlobalEgressIP)
+	numberOfIPs := clusterGlobalEgressIP.Spec.NumberOfIPs
+	if numberOfIPs == nil {
+		one := 1
+		numberOfIPs = &one
+	}
```

Comment:
```suggestion
	numberOfIPs := 1
	if clusterGlobalEgressIP.Spec.NumberOfIPs != nil {
		numberOfIPs = *clusterGlobalEgressIP.Spec.NumberOfIPs
	}
```

And from this point on you can avoid passing down pointers, because I was reading the other functions and suffering because numberOfIps could be nil, although you set it to &one here ;)